modules/network/subnets.tf

```diff
@@ -79,7 +79,7 @@ locals {
   # - Subnet is configured with newbits and/or netnum/cidr
   # - Not configured with create == 'never'
   # - Not configured with an existing 'id'
-  subnets_to_create = length(var.vcn_cidrs) > 0 ? merge(
+  subnets_to_create = merge(
```

Comment:
No reason to safeguard against an empty `var.vcn_cidrs` list, the plan will already fail elsewhere if `length(var.vcn_cidrs) == 0`

Reply:
Not sure if it's the case here, but some defensive coding in the project has related to errors on destroy, such as missing inputs that shouldn't matter blocking cleanup of known state, and we may need to handle the error vs. fail elsewhere.

Reply:
@devoncrouse Thanks for looking through this PR. Understand the potential issue, any idea of a potential error case I can try to reproduce to see what workarounds are possible? Without a bit more context, one failsafe i can think of is to replace
```
subnets_to_create = length(var.vcn_cidrs) > 0 ? merge(...) : {}
```
with 
```
subnets_to_create = try(merge(...), {})
```
(assuming the issue is that calculating `subnets_to_create` throws an error when `var.vcn_cidrs` is empty).
This avoids `local.subnets_to_create` from having a dependency on `data.oci_core_vcn.oke` from the root module, which is where this problem comes from.

Let me know thoughts, thanks!

Reply:
@chrizkim :+1: that seems safe, thanks!

Reply:
done

Reply:
This still fails if, in the `subnets `variable, we are providing the `newbits `instead of the `cidr` attributes.

Here is an example of the subnets variable that can be used for test:

```
    subnets = {
        bastion  = { newbits = 13 }
        operator = { newbits = 13 }
        cp       = { newbits = 13 }
        int_lb   = { newbits = 11 }
        pub_lb   = { newbits = 11 }
        workers  = { newbits = 2 }
        pods     = { newbits = 2 }
        # bastion  = { cidr = "10.0.0.0/29" }
        # operator = { cidr = "10.0.0.8/29" }
        # cp       = { cidr = "10.0.0.16/29" }
        # int_lb   = { cidr = "10.0.0.64/27" }
        # pub_lb   = { cidr = "10.0.0.128/27" }
        # workers  = { cidr = "10.0.8.0/21" }
    }
```


To work around the problem with `newbits` is necessary to remove one validation from the `subnets_to_create` variable:

`# contains(keys(local.subnet_cidrs_all), k),                                              # has a calculated CIDR range (not id input)`